### PR TITLE
added back pre-main-kyma-gardener-gcp-eventing with kyma deploy

### PR DESCRIPTION
**Pull request by mfaizanse**
**Description**

Changes proposed in this pull request:

- added back pre-main-kyma-gardener-gcp-eventing
- Changed job to use Kyma deploy instead of kyma install
- Changed to use Eventing specific E2E fast-integration tests

[Example Job run](https://status.build.kyma-project.io/view/gs/kyma-prow-logs/pr-logs/pull/kyma-project_kyma/12156/mfaizanse_test_of_prowjob_pre-main-kyma-gardener-gcp-eventing/1442415497755758592)

**Related issue(s)**

- Dependent on PR: https://github.com/kyma-project/kyma/pull/12156

### CODE REVIEW

File: prow/scripts/cluster-integration/kyma-integration-gardener-eventing.sh [hunk 1]
```diff
@@ -119,11 +121,11 @@ gardener::generate_overrides
 
 gardener::provision_cluster
 
-log::banner "Create EventMesh Secret"
-eventing::create_eventmesh_secret
```
Comment (mfaizanse) [inline prow/scripts/cluster-integration/kyma-integration-gardener-eventing.sh:123]: Is this secret needed anywhere?

File: prow/jobs/kyma/kyma-integration-gardener.yaml [hunk 2]
```diff
@@ -227,6 +227,78 @@ presubmits: # runs on PRs
             effect: NoSchedule
         nodeSelector:
             dedicated: "high-cpu"
+    - name: pre-main-kyma-gardener-gcp-eventing
+      annotations:
+        pipeline.clusterprovisioning: "kyma cli"
+        pipeline.installer: "kyma delpoy"
+        pipeline.platform: "gardener_gcp"
+        pipeline.test: "octopus"
```
Comment (mfaizanse) [inline prow/jobs/kyma/kyma-integration-gardener.yaml:235]: Should I change it to "fast-integration"?

File: templates/data/kyma-integration-gardener-data.yaml [hunk 3]
```diff
@@ -38,8 +38,8 @@ templates:
             command: "/home/prow/go/src/github.com/kyma-project/test-infra/prow/scripts/cluster-integration/kyma-integration-gardener.sh"
           command_integration_eventing:
             annotations:
-              pipeline.installer: kyma install
-              pipeline.test: octopus
+              pipeline.installer: kyma delpoy
```
[the remark this answers is not in the document]
Comment (mfaizanse) [reply templates/data/kyma-integration-gardener-data.yaml]: fixed

**Review by sayanh — APPROVED** [review]

**Review by alexandra-simeonova — APPROVED** [review]

**Pull request merged**